lib/bugsnag/integrations/sidekiq.rb

```diff
@@ -24,9 +24,10 @@ def call(worker, msg, queue)
         yield
       rescue Exception => ex
         self.class.notify(ex) unless self.class.sidekiq_supports_error_handlers
+        keep_request_data = self.class.sidekiq_supports_error_handlers
         raise
       ensure
-        Bugsnag.configuration.clear_request_data
+        Bugsnag.configuration.clear_request_data unless keep_request_data
```

Comment:
We should just use `self.class.sidekiq_supports_error_handlers`. This is not defined in the case where there is no exception currently which is confusing and buggy looking.

Reply:
we could have a boolean that is `error_raised` that is set to false initially then tru in the rescue. Then also we could have the if statement ammended to check both conditions with a comment explaining why.

Reply:
Ok, if that makes it more clear